### ISSUE TITLE
unban members not in room, strange behaviour

### ISSUE THREAD

**Issue by theotheroracle** [vector-im/element-web#16822]
![Screen Shot 2021-03-30 at 4 13 21 AM](https://user-images.githubusercontent.com/20211219/112964859-51084c00-910e-11eb-8f71-0b240d30449e.png)

### Description

element allows you to both unban members that are not in the room and displays the event as if they were kicked.

### Steps to reproduce

- do /unban  @whatever:you.want ( user can't have been in the room before )

Describe how what happens differs from what you expected.

Logs being sent: yes/no

### Version information

- **Platform**: web

For the web app:

- **Browser**: Firefox
- **OS**: macOS
- **URL**: tested on element.io

**Comment by ShadowJonathan** [comment]
This is because unbanning effectively changes the user state from `ban` -> `leave`, which means that element subsequently sees `*` -> `leave`, which is "to kick", which shows this.

Effectively, element just sends a `leave` event for the user in question when doing an `/unban`, i think element should check that user's state first before doing this, because it can result in situations such as this.

**Comment by theotheroracle** [comment]
yeah thats what i meant by "if there's no state for the user it should say so"
